### PR TITLE
Add tests for ExtractMethod on deconstruction. Test utility to output full actual result.

### PR DESCRIPTION
**Pull request by jcouv**
It looks like refactoring issues with declaration expressions have been fixed already (likely by PR https://github.com/dotnet/roslyn/pull/15646). So I just added some tests.

Closes https://github.com/dotnet/roslyn/issues/14108

### CODE REVIEW

**Comment by jcouv** [comment]
windows_build_correctness_prtest failed with error below ([details](https://ci.dot.net/job/dotnet_roslyn/job/master/job/windows_build_correctness_prtest/8628/))

```
15:39:18 Command failed to execute: C:\Users\dotnet-bot\.nuget\packages\RoslynTools.MSBuild.0.3.0-alpha\tools\msbuild\msbuild.exe /noconlog /v:m /m /p:Configuration=Debug /logger:StructuredLogger,C:\Users\dotnet-bot\.nuget\packages\Microsoft.Build.Logging.StructuredLogger\1.1.130\lib\net46\StructuredLogger.dll;D:\j\workspace\windows_build---1b9ec577\Binaries\Debug\roslyn.buildlog /nodeReuse:false /p:DeployExtension=false D:\j\workspace\windows_build---1b9ec577\Roslyn.sln
15:39:18 System.Management.Automation.RuntimeException: Command failed to execute: C:\Users\dotnet-bot\.nuget\packages\RoslynTools.MSBuild.0.3.0-alpha\tools\msbuild\msbuild.exe /noconlog /v:m /m /p:Configuration=Debug /logger:StructuredLogger,C:\Users\dotnet-bot\.nuget\packages\Microsoft.Build.Logging.StructuredLogger\1.1.130\lib\net46\StructuredLogger.dll;D:\j\workspace\windows_build---1b9ec577\Binaries\Debug\roslyn.buildlog /nodeReuse:false /p:DeployExtension=false D:\j\workspace\windows_build---1b9ec577\Roslyn.sln
15:39:18 Command failed to execute:  & ".\build\scripts\test-build-correctness.ps1" -config $buildConfiguration 
15:39:18 System.Management.Automation.RuntimeException: Command failed to execute:  & ".\build\scripts\test-build-correctness.ps1" -config $buildConfiguration 
15:39:18 at Exec-Block, D:\j\workspace\windows_build---1b9ec577\build\scripts\build-utils.ps1: line 26
15:39:18 at Test-Special, D:\j\workspace\windows_build---1b9ec577\build\scripts\build.ps1: line 236
15:39:18 at <ScriptBlock>, D:\j\workspace\windows_build---1b9ec577\build\scripts\build.ps1: line 524
15:39:18 Killing running vs processes...
```

**Comment by jcouv** [comment]
test windows_build_correctness_prtest  please

**Comment by jcouv** [comment]
test windows_debug_vs-integration_prtest  please

**Comment by jcouv** [comment]
test windows_release_vs-integration_prtest  please

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for review. Thanks

File: src/Test/Utilities/Portable/Syntax/TokenUtilities.cs [hunk 1]
```diff
@@ -42,7 +42,7 @@ public static class TokenUtilities
                         }
                     }
 
-                    AssertEx.Fail($"Unexpected token.  Actual '{actualAll}' Expected '{expectedAll}'");
+                    AssertEx.Fail($"Unexpected token.  Actual '{actualAll}' Expected '{expectedAll}'\r\nActual:\r\n{actual}");
```
Comment (jasonmalinowski) [inline src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: So there are two actuals? This seems confusing.

Comment (jcouv) [inline src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: @jasonmalinowski Here's what the output looks like. I don't think it's confusing, but I'm happy to mark the second one as "Actual (full):" if that helps.

```
Test 'Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings.ExtractMethod.ExtractMethodTests.TestDeconstruction5' failed:
	Unexpected token.  Actual 'private static void ^NewMethod^ ( out int x , ' Expected 'private static void ^NeMethod^ ( out int x , '
Actual:
class Program
{
    void M()
    {
        int x, y;
        NewMethod(out x, out y);
        System.Console.Write(x + y);
    }

    private static void NewMethod(out int x, out int y)
    {
        (x, y) = (1, 2);
    }
}
Expected: False
Actual:   True
```

**Comment by jcouv** [comment]
test windows_release_vs-integration_prtest please

**Comment by jcouv** [comment]
@jasonmalinowski @roslyn-ide Adding tests and small change to test helper. Let me know if ok. Thanks

File: src/Test/Utilities/Portable/Syntax/TokenUtilities.cs (hunk 1 above)
Comment (jcouv) [inline src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: @jasonmalinowski Does that make sense, or should I change it?

Comment (jcouv) [reply src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: @jasonmalinowski Does that make sense, or should I change it?

Comment (jcouv) [reply src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: @jasonmalinowski Are you ok with this test output, despite "Actual" appearing multiple times?

Comment (jasonmalinowski) [reply src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: @jcouv I'm still completely unable to explain what it means. Why do we need two?

Comment (jcouv) [reply src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: @jasonmalinowski The existing assert only prints a few tokens. This is convenient when you have a small difference (it pinpoints the difference). 
But when you have a large difference (let's say you're writing a new test), that is not very useful. The message I added is useful then. 

I don't think the printout is confusing. Let me know what you want to do: change labels (use `"Actual (full):"`), remove the new section, or something else entirely ;-)

Comment (jcouv) [reply src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: For context, see the example I copied above.

Comment (jcouv) [reply src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: Latest commit removes some waste. Here's what the new output looks like:

```
Test 'Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings.ExtractMethod.ExtractMethodTests.TestDeconstruction5' failed:
	Unexpected token.  Actual 'y ) ; ^System^ . Console . Write ( ' Expected 'y ) ; ^}^ private static void NewMethod ( '
Actual:
class Program
{
    void M()
    {
        int x, y;
        NewMethod(out x, out y);
        System.Console.Write(x + y);
    }

    private static void NewMethod(out int x, out int y)
    {
        (x, y) = (1, 2);
    }
}
	at Roslyn.Test.Utilities.AssertEx.Fail(String message)
	at Roslyn.Test.Utilities.TokenUtilities.AssertTokensEqual(String expected, String actual, String language)
	at Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions.AbstractCodeActionOrUserDiagnosticTest.<TestOperationsAsync>d__24.MoveNext()
	--- End of stack trace from previous location where exception was thrown ---
	at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
    ....
```

Comment (jcouv) [reply src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: @jasonmalinowski Good to go?

Comment (jasonmalinowski) [reply src/Test/Utilities/Portable/Syntax/TokenUtilities.cs]: Oh sorry, yeah, ship it.

**Pull request merged**